pkg/apis/monitoring/v1/types.go

```diff
@@ -1626,3 +1641,37 @@ func (l *PrometheusRuleList) DeepCopyObject() runtime.Object {
 type ProbeTLSConfig struct {
 	SafeTLSConfig `json:",inline"`
 }
+
+// Authorization contains optional `Authorization` header configuration.
+// This section is only understood by  versions of Prometheus >= 2.26.0.
+// +k8s:openapi-gen=true
+type Authorization struct {
+	// Set the authentication type. Defaults to Bearer
+	Type string `json:"type,omitempty"`
+	// Secret to read or mount
+	Credentials *v1.SecretKeySelector `json:"credentials,omitempty"`
+	// File to read a secret from
+	CredentialsFile string `json:"credentialsFile,omitempty"`
+}
+
+// Validate semantically validates the given Authorization section.
+func (c *Authorization) Validate() error {
+	if c.Credentials != nil && c.CredentialsFile != "" {
+		return &AuthorizationValidationError{"Authorization can not specify both Credentials and CredentialsFile"}
+	}
+	if strings.ToLower(strings.TrimSpace(c.Type)) == "basic" {
```

Comment:
What is the a reason for explicitly disallowing the `basic` type? If the reason is the dedicated section for basic auth, I would still think it's fine to allow it here. Nothing would really break if we allowed it, and we could also add a section in the docs that the basic type is already supported in a different way.

Reply:
[Prometheus doesn't allow it](https://github.com/prometheus/common/blob/main/config/http_config.go#L218) and we'd end up with a invalid config.

Reply:
Ack, wasn't aware of that